# Calvin Forinash - Raw Data and Analysis #

In this notebook, I create my DataFrame and perform some basic analysis of crime reporting numbers to find the "safest" neighborhoods in Pittsburgh. Obviously, crime reports are simply not enough to say that a neighborhood is safe, since I won't be looking at what the offenses were. Some offenses are bound to be minor, while some will be pretty major crimes; these would both be worth the same value in this analysis. However, combined with Ola and Matthew's analyses and their metrics, I believe that crime statistics should help determine which neighborhoods would be good to live in. 

In [2]:
import pandas as pd
import numpy as np

data = pd.read_csv("ctf16_crime_data.csv")
data.head()

PK         CCR  HIERARCHY         INCIDENTTIME  \
0  2802309  16000001.0         10  2016-01-01T00:00:00   
1  2803174  16004547.0         11  2016-01-01T00:01:00   
2  2801809  16000367.0          4  2016-01-01T00:10:00   
3  2802315  16000035.0         10  2016-01-01T00:15:00   
4  2802312  16000024.0          4  2016-01-01T00:16:00   

                                INCIDENTLOCATION CLEAREDFLAG  \
0  400 Block North Shore DR Pittsburgh, PA 15212           Y   
1    5400 Block Carnegie ST Pittsburgh, PA 15201           N   
2  500 Block Mt Pleasant RD Pittsburgh, PA 15214           N   
3         300 Block Wood ST Pittsburgh, PA 15222           Y   
4  500 Block Mt Pleasant RD Pittsburgh, PA 15214           N   

          INCIDENTNEIGHBORHOOD INCIDENTZONE  \
0                  North Shore            1   
1          Upper Lawrenceville            2   
2            Northview Heights            1   
3  Golden Triangle/Civic Arena            2   
4            Northview Heights            1   

                      INCIDENTHIERARCHYDESC  \
0            HARRASSMENT/THREAT/ATTEMPT/PHY   
1                        THEFT BY DECEPTION   
2   DISCHARGE OF FIREARM INTO OCC.STRUCTURE   
3            HARRASSMENT/THREAT/ATTEMPT/PHY   
4  PROP MISSILE INTO OCC VEHICLE/OR ROADWAY   

                                            OFFENSES  INCIDENTTRACT  \
0  2702 Aggravated Assault. / 2709(a) Harassment....         2205.0   
1                           3922 Theft by Deception.         1011.0   
2  2707.1 Discharge of a Firearm into Occupied St...         2609.0   
3        2709(a)(3) Harassment No Legitimate Purpose          201.0   
4  2705 Recklessy Endangering Another Person. / 3...         2609.0   

   COUNCIL_DISTRICT  PUBLIC_WORKS_DIVISION          X          Y  
0               1.0                    6.0 -80.012337  40.446263  
1               7.0                    2.0 -79.950295  40.482290  
2               1.0                    1.0 -80.000966  40.478651  
3               6.0                    6.0 -80.001251  40.438918  
4               1.0                    1.0 -80.000966  40.478651

Below, I will analyze this data and put together a list of the top 10 safest neighborhoods in Pittsburgh based on these crime statistics. 

In [3]:
data['INCIDENTNEIGHBORHOOD']

0                         North Shore
1                 Upper Lawrenceville
2                   Northview Heights
3         Golden Triangle/Civic Arena
4                   Northview Heights
                     ...             
239500                       Sheraden
239501                      Brookline
239502                    South Shore
239503              Northview Heights
239504                      Allentown
Name: INCIDENTNEIGHBORHOOD, Length: 239505, dtype: object

The next data is for Central and South Oakland, just for fun. It's relatively safe, but by no means on the low end of crime statistics. Since 2016 there have been 1692 UCR reports in South Oakland. Central Oakland has seen 4033 reports, I assume mostly because of stupid college students getting in trouble. 

In [25]:
#data['INCIDENTNEIGHBORHOOD'].count()
southO = data.loc[data['INCIDENTNEIGHBORHOOD'] == 'South Oakland']
centralO = data.loc[data['INCIDENTNEIGHBORHOOD'] == 'Central Oakland']

centralO

PK         CCR  HIERARCHY         INCIDENTTIME  \
56      2801506  16000227.0         24  2016-01-01T04:16:00   
58      2801508  16000233.0         14  2016-01-01T04:30:00   
285     2801273  16001601.0          5  2016-01-03T19:30:00   
375     2801891  16002190.0         99  2016-01-04T19:06:00   
395     2801958  16002318.0          3  2016-01-04T23:45:00   
...         ...         ...        ...                  ...   
239217  3218869  21051191.0         99  2021-03-26T20:35:00   
239314  3219021  21052151.0         23  2021-03-28T04:19:00   
239336  3219051  21051348.0         99  2021-03-27T01:35:00   
239425  3219216  21052873.0          6  2021-03-29T12:30:00   
239438  3219233  21052919.0         99  2021-03-29T15:10:00   

                                     INCIDENTLOCATION CLEAREDFLAG  \
56         Meyran AV & Forbes AV Pittsburgh, PA 15213           Y   
58          400 Block Oakland AV Pittsburgh, PA 15213           N   
285          300 Block Mc Kee PL Pittsburgh, PA 15213           N   
375     S Bouquet ST & Forbes AV Pittsburgh, PA 15213           N   
395         300 Block Oakland AV Pittsburgh, PA 15213           N   
...                                               ...         ...   
239217       Oakland AV & 5th AV Pittsburgh, PA 15213           N   
239314       300 Block Semple ST Pittsburgh, PA 15213           Y   
239336    Forbes AV & Oakland AV Pittsburgh, PA 15213           N   
239425       10 Block Oakland SQ Pittsburgh, PA 15213           N   
239438                  Melba PL Pittsburgh, PA 15213           Y   

       INCIDENTNEIGHBORHOOD INCIDENTZONE  INCIDENTHIERARCHYDESC  \
56          Central Oakland            4     DISORDERLY CONDUCT   
58          Central Oakland            4      CRIMINAL MISCHIEF   
285         Central Oakland            4  BURGLARY/NO FORCE/RES   
375         Central Oakland            4                    NaN   
395         Central Oakland            4    ROBBERY/HIGHWAY/GUN   
...                     ...          ...                    ...   
239217      Central Oakland            4                    NaN   
239314      Central Oakland            4      PUBLIC DRUNKENESS   
239336      Central Oakland            4                    NaN   
239425      Central Oakland            4         THEFT BICYCLES   
239438      Central Oakland            4                    NaN   

                                                 OFFENSES  INCIDENTTRACT  \
56      5503 Disorderly Conduct. / 7728 Accidents and ...          405.0   
58                                         3502 Burglary.          406.0   
285                                        3502 Burglary.          405.0   
375     3743 Accidents Involving Damage to Attended Ve...          405.0   
395                                         3701 Robbery.          405.0   
...                                                   ...            ...   
239217  3745 Accidents Involving Damage to Unattended ...          405.0   
239314  5505 Public Drunkenness / 5506 Loitering and P...          405.0   
239336  3745 Accidents Involving Damage to Unattended ...          405.0   
239425      3921 Theft by Unlawful Taking or Disposition.          406.0   
239438                                   9994 PFA Service          406.0   

        COUNCIL_DISTRICT  PUBLIC_WORKS_DIVISION          X          Y  
56                   3.0                    3.0 -79.958317  40.440579  
58                   3.0                    3.0 -79.953489  40.438644  
285                  3.0                    3.0 -79.957719  40.438309  
375                  3.0                    3.0 -79.956438  40.441949  
395                  3.0                    3.0 -79.954990  40.439887  
...                  ...                    ...        ...        ...  
239217               3.0                    3.0 -79.957857  40.442219  
239314               3.0                    3.0 -79.957092  40.438809  
239336               3.0                    3.0 -79.956974  4

In [22]:
sorted_data = data['INCIDENTNEIGHBORHOOD'].value_counts()
sorted_data.head(15)

South Side Flats               13972
Central Business District      12146
Carrick                         8455
Bloomfield                      6563
Shadyside                       6249
East Liberty                    5658
Squirrel Hill South             5436
Homewood South                  5359
Mount Washington                5228
Brookline                       5226
Lincoln-Lemington-Belmar        4831
Knoxville                       4713
Homewood North                  4537
Brighton Heights                4526
Golden Triangle/Civic Arena     4278
Name: INCIDENTNEIGHBORHOOD, dtype: int64

## Least safe neighborhoods ##
Based on this data, the 3 neighborhoods with the most UCR (Uniform Crime Reporting) reports are South Side Flats, the Central Business District and Carrick. 
1. **South Side Flats** - 13972
* This neighborhood sits on the south bank of the Monongahela River, almost directly south of the University of           Pittsburgh. It includes landmarks such as the Color Park and walking trails along the river. The neighborhood is also     centered around E Carson St, which has over 40 bars and pubs; this could be a contributing factor to high crime           reports. 

2. **Central Business District (aka Downtown)** - 12146
* Downtown sits between the Monongahela and Allegheny Rivers. Home to bars, skyscrapers, malls, good food and sports arenas, it is unsurprising that the business center of a historically industrial city is also home to high crime rates. This could also be due simply to the huge amount of traffic, both vehicular and pedestrian, that the neighborhood attracts; more people will eventually lead to more crime. Downtown is often colloquially referred to as The Golden Triangle, which is included separately and has the 15th most UCR reports with 4278. Including this in the Central Business District's tally makes it the neighborhood with the most reports with 16,424, over 3,000 more than any other neighborhood.

3. **Carrick** - 8455
* Everything after South Side Flats and Downtown has significantly fewer crime reports——at least 3700 fewer over the last 5 years. However, Carrick reported almost 2000 more crimes than the next highest, Bloomfield. Carrick sits on the South Side of Pittsburgh. As a densely packed neighborhood in an industrial city, it makes sense that it is a crime hotspot. Based on a quick Google search for "Carrick Pittsburgh" that immediately turned up articles from a few days ago (4/11) about a home invasion in which two men were assaulted. 

However, I want to find the *safest* neighborhoods. There are a handful of neighborhoods with fewer than 600 reports over the last 5+ years (this dataset spans from 2016-present), meaning that they generally get under 100 reports every year. The list below lists the 20 neighborhoods that have officially reported the fewest crimes since 2016. 

In [26]:
reverse_sorted_data = data['INCIDENTNEIGHBORHOOD'].value_counts().iloc[::-1]
reverse_sorted_data.head(25)

Mt. Oliver Boro             72
Mt. Oliver Neighborhood    117
Outside County             154
Outside State              218
Chartiers City             222
New Homestead              248
Ridgemont                  248
Troy Hill-Herrs Island     252
Swisshelm Park             318
East Carnegie              333
Arlington Heights          367
Mount Oliver               377
Summer Hill                394
Hays                       401
Regent Square              408
Oakwood                    439
Esplen                     454
Glen Hazel                 516
Fairywood                  558
St. Clair                  584
Windgap                    604
Spring Garden              679
Polish Hill                737
Duquesne Heights           783
Central Northside          786
Name: INCIDENTNEIGHBORHOOD, dtype: int64

## Safest Neighborhoods by the numbers ##
Based on this data, the 5 safest (actual) neighborhoods are Mount Oliver, Chartiers, New Homestead, Ridgemont and Troy Hill/Herrs Island. I will only go in depth into the top 3 lowest crime neighborhoods. 

1. **Mount Oliver** - 189
* Mount Oliver is, coincidentally, directly north of Carrick. Although the neighborhood of Mt Oliver is  distinct from Mt Oliver Borough, I am considering them as one neighborhood (that technically isn't part of the city of Pittsburgh). It is a relatively high-end residential neighborhood. According to Wikipedia, the average household size is 2.36, which indicates that its residents are largely couples or single parents, with the occasional large family or person living alone. 

2. **Chartiers** - 222
* Chartiers is a small neighborhood in the northwest corner of Pittsburgh. With a land area of a mere 84 acres (0.132 sq. miles) and population of less than 500, the crime rate per capita is relatively high. I won't include this neighborhood on the final top 10 list I compile. 

3. **New Homestead** - 248
* Mostly encompassing wooded areas, New homestead has a population of about 1000 residents in an area of .794 sq miles (again, I won't include this in the final list). 

4. **Ridgemont** - 248

5. **Troy Hill/Herrs Island** - 252

## Let's Talk About Mount Oliver ##

In [37]:
mtoliver = data.loc[data['INCIDENTNEIGHBORHOOD'] == 'Mt. Oliver Boro']
mtoliver.head()

PK         CCR  HIERARCHY         INCIDENTTIME  \
4933   2811938  16029663.0         14  2016-02-18T07:50:00   
5608   2812987  16033854.0         11  2016-02-24T12:00:00   
7408   2817610  16044545.0         99  2016-03-12T04:30:00   
9578   2825111  16061606.0         99  2016-04-01T07:00:00   
14915  2831789  16090396.0         10  2016-05-17T14:05:00   

                                        INCIDENTLOCATION CLEAREDFLAG  \
4933       300 Block Brownsville RD Pittsburgh, PA 15210           N   
5608       400 Block Brownsville RD Pittsburgh, PA 15210           N   
7408   Liberty TUN & E Warrington AV Pittsburgh, PA 1...           N   
9578           200 Block ANTHONY ST PITTSBURGH, PA 15210           N   
14915      700 Block Brownsville RD Pittsburgh, PA 15210           N   

      INCIDENTNEIGHBORHOOD INCIDENTZONE           INCIDENTHIERARCHYDESC  \
4933       Mt. Oliver Boro            3               CRIMINAL MISCHIEF   
5608       Mt. Oliver Boro            3               THEFT OF SERVICES   
7408       Mt. Oliver Boro            3                             NaN   
9578       Mt. Oliver Boro            3                             NaN   
14915      Mt. Oliver Boro            3  HARRASSMENT/THREAT/ATTEMPT/PHY   

                                                OFFENSES  INCIDENTTRACT  \
4933                             3304 Criminal Mischief.         4810.0   
5608                9999 Miscellaneous Report (No Crime)         4810.0   
7408   3743 Accidents Involving Damage to Attended Ve...         4810.0   
9578                                  9999 (Misc. Crime)         4810.0   
14915  2709(a)(1) Harassment by Physical Contact, or ...         4810.0   

       COUNCIL_DISTRICT  PUBLIC_WORKS_DIVISION          X          Y  
4933                3.0                    4.0 -79.988359  40.414290  
5608                3.0                    4.0 -79.989126  40.413245  
7408                NaN                    NaN   0.000000   0.000000  
9578                NaN                    NaN -79.986476  40.412247  
14915               3.0                    4.0 -79.990902  40.409848

Looking at the offenses column, most of the crimes committed in Mt Oliver are relatively minor, with a couple exceptions, notably the terroristic threats

## Final List - Safest Neighborhoods ##

Below is the final top 20 list of the safest neighborhoods. I won't go into detail on every one so that I can put together as long and comprehensive a list as possible to make the combination of metrics with my groupmates easier.
  (*Keep in mind, safest is subjective here, and I included and omitted some neighborhoods based on population or other factors that I felt discounted their statistics*)

#### 1. Mount Oliver ####
* Based on my analysis, Mount Oliver is my pick for the safest neighborhood in Pittsburgh. It only had 189 UCR reports since 2016, while having a reasonable population of about 4000 residents. 

#### 2. Chartiers ####

#### 3. New Homestead ####

#### 4. Ridgemont ####

#### 5. Troy Hill/Herrs Island ####

#### 6. Swisshelm Park ####

#### 7. East Carnegie ####

#### 8. Arlington Heights ####

#### 9. Summer Hill ####

#### 10. Hays ####

#### 11. Regent Square ####

#### 12. Oakwood ####

#### 13. Esplen ####

#### 14. Glen Hazel ####

#### 15. Fairywood ###

#### 16. Windgap ####

#### 17. Spring Garden ####

#### 18. Polish Hill ####

#### 19. Duquesne Heights ####

#### 20. Central Northside ####

## Interesting Findings ##

* A fair few of the safest neighborhoods are in far east (Swisshelm Park, New Homestead, Regent Square, Glen Hazel) or northwest Pittsburgh (Chartiers, Esplen, Fairywood, Windgap). This makes intuitive sense to me; the farther away from downtown they are, the less traffic they get, and the less likely to have high crime rates. 

* Western and southwestern Pittsburgh features Ridgemont, East Carnegie, Oakwood. This is interesting considering that many of the highest crime neighborhoods are also found in southwestern Pittsburgh. 

* Polish Hill, despite being pretty close to downtown, separated only by the Strip and Hill Districts, has experienced relatively little crime recently. Originally home to Polish immigrants and later thei descendants, it sounds about right that it is safe. Although immigrants get a bad rap when it comes to crime, these immigrants are hard working, dedicated individuals who would rather not sacrifice their shot at the "American Dream" and disregard or throw away the hardship their ancestors endured to get there in the first place. 